lib/diagnostic.rb

```diff
@@ -6,6 +6,10 @@ class Diagnostic
   # method that returns a Fixnum.
   def longest(enum)
     # Your code goes here
+    longest = enum.reduce do |memo, obj|
+      memo.length > obj.length ? memo : obj
+    end
```

Comment:
This is one of several ways that you could get the desired behavior. However, there are easier options: for instance, `max` or `max_by`. Here's how they get called.
`enum.max_by { |entry| entry.length }` or `enum.max_by(&:length)`
`enum.max { |a.b| a.length <=> b.length }`